generator/README.md

```diff
@@ -0,0 +1,824 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+```
+
+Where `[year]` in the copyright line is the current (?) year.
+
+### General rules for Objective-C classes
+1. Default initializer applies only to Functions(Request / Response / Notification) classes
+
+```objc
+    - (instancetype)init {
+        self = [super initWithName:SDLRPCFunctionNameRegisterAppInterface];
+        if (!self) { return nil; }
+
+        return self;
+    }
+```
+*Pease note the required double brackets in the if statement*
+
+2. Initializer for mandatory params if there is/are any in XML (skipped if no <mandatory = true> params)
+3. Initializer for all params if there is/are any which is not mandatory in XML (skipped if no <mandatory = false> params)
+
+#### Scalars
+There are 4 type of scalar values declared in the SDL lib.  These are:
+1. **SDLInt** - A declaration that this NSNumber contains an NSInteger.
+0. **SDLUInt** - A declaration that this NSNumber contains an NSUInteger.
+0. **SDLBool** - A declaration that this NSNumber contains a BOOL.
+0. **SDLFloat** - A declaration that this NSNumber contains a float.
+*Note: These are syntactic sugar to help the developer know what type of value is held in the `NSNumber`.*
+
+```objc
+@interface NSNumber (NumberType) <SDLInt, SDLUInt, SDLBool, SDLFloat>
+```
+Usage example:
+```objc
+@property (strong, nonatomic) NSNumber<SDLInt> *touchEventId;
+```
+or an array:
+```objc
+@property (strong, nonatomic) NSArray<NSNumber<SDLInt> *> *timeStamp;
+```
+
+#### Enums
+RPC Enums in SDL are strings. sdl_ios uses `NSString` `typedef`ed with a proper enum type. In Swift projects, however, they become real enums by using the `NS_SWIFT_ENUM` compiler tag.
+
+*Base definition of `SDLEnum`:*
+
+```objc
+typedef NSString* SDLEnum SDL_SWIFT_ENUM;
+
+*Note: This new defined type has already adds a pointer, so anything that inherits from `SDLEnum` needs no asterisk.*
+
+```objc
+typedef SDLEnum SDLTouchType SDL_SWIFT_ENUM; // SDLTouchType will be considered an NSString by the compiler in Obj-C, but will be an enum object of type SDLTouchType in Swift.
+```
+
+And here is a concrete 'enum' item:
+
+```objc
+extern SDLTouchType const SDLTouchTypeBegin;
+```
+
+If an item is deprecated then it will be declared as such:
+
+```objc
+__deprecated
+extern SDLTouchType const SDLTouchTypeBegin;
```

Comment:
This is not show it should be generated. It should be generated as:

```objc
extern SDLTouchType const SDLTouchTypeBegin __deprecated;
```

Reply:
@joeljfischer please have look into commit below, if understand you right  https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/0f53ab92a5589ee857aa96104f6f7c34e319e4fc

Reply:
I don't think that that's correct. If I'm understanding this code change correctly, you're changing the `@interface` definition to have `__deprecated` next to it instead of above it. That's not correct and will cause a build error. For `@interface` definitions, it should be above it.

Not:

```objc
@interface SDLRPCMessage : SDLRPCStruct <NSCopying> __deprecated
```

Instead:

```objc
__deprecated
@interface SDLRPCMessage : SDLRPCStruct <NSCopying>
```

However, for parameters or for constants (like the one referenced in the initial comment) it should be next to it like I described in my first comment. Does that make sense?

Reply:
Basically it should be above a type definition and next to a parameter or constant. I can't currently think of any other cases.

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/45f8005b9c7bd32e72f8c0891e85feb55d688d93